From 3b3c7f93a8ee4110b4834d4582666ce9b6302de9 Mon Sep 17 00:00:00 2001
From: Takuto Ikuta <tikuta@chromium.org>
Date: Fri, 28 May 2021 04:02:24 +0000
Subject: [PATCH] Reland "components/resources: use python3 for
 //tools/licenses.py"

This reverts commit 18028c51b35dbe4e54915366f28cd2f915e9a81b.

Reason for revert:
Fix non-determinisity around set()

Original change's description:
> Revert "components/resources: use python3 for //tools/licenses.py"
>
> This (partially) reverts commit a7febb8cb85a359825f909b398401d810ba313f4.
>
> Reason for revert: Causes non-determinism
>
> Original change's description:
> > components/resources: use python3 for //tools/licenses.py
> >
> > Specify encoding for read failure on Windows, and use codecs.open as
> > builtin open doesn't have encoding args in python2.
> >
> > Fixed: 1205601
> > Change-Id: I8fd2ea76f3b4ee443c437249aa371fdf34f5776b
> > Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/2905676
> > Auto-Submit: Takuto Ikuta <tikuta@chromium.org>
> > Commit-Queue: Takuto Ikuta <tikuta@chromium.org>
> > Reviewed-by: Nico Weber <thakis@chromium.org>
> > Cr-Commit-Position: refs/heads/master@{#884787}
>
> Bug: 1205601, 1211349
> Change-Id: I984726b2388cff3994c8221408173f9b346c2cdf
> Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/2920328
> Auto-Submit: Andrew Grieve <agrieve@chromium.org>
> Commit-Queue: Takuto Ikuta <tikuta@chromium.org>
> Reviewed-by: Nico Weber <thakis@chromium.org>
> Reviewed-by: Takuto Ikuta <tikuta@chromium.org>
> Cr-Commit-Position: refs/heads/master@{#886984}

Fixed: 1205601
Fixed: 1211349
Change-Id: I90ee09960d1b56f59f1922bb04bde960a0c0de97
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/2921665
Auto-Submit: Takuto Ikuta <tikuta@chromium.org>
Commit-Queue: Nico Weber <thakis@chromium.org>
Reviewed-by: Nico Weber <thakis@chromium.org>
Cr-Commit-Position: refs/heads/master@{#887435}
---
 src/components/resources/BUILD.gn | 4 +---
 src/tools/licenses.py             | 4 ++--
 2 files changed, 3 insertions(+), 5 deletions(-)

diff --git a/src/components/resources/BUILD.gn b/src/components/resources/BUILD.gn
index fb82ba6431..9b87aa18ff 100644
--- a/src/components/resources/BUILD.gn
+++ b/src/components/resources/BUILD.gn
@@ -3,7 +3,6 @@
 # found in the LICENSE file.
 
 import("//build/config/android/config.gni")
-import("//build/config/python.gni")
 import("//components/safe_browsing/buildflags.gni")
 import("//printing/buildflags/buildflags.gni")
 import("//tools/grit/grit_rule.gni")
@@ -84,8 +83,7 @@ grit("components_scaled_resources") {
   output_dir = "$root_gen_dir/components"
 }
 
-# TODO(crbug.com/1112471): Get this to run cleanly under Python 3.
-python2_action("about_credits") {
+action("about_credits") {
   script = "//tools/licenses.py"
   depfile = "$target_gen_dir/$target_name.d"
 
diff --git a/src/tools/licenses.py b/src/tools/licenses.py
index aaa520b1ae..a1e1f9afb3 100755
--- a/src/tools/licenses.py
+++ b/src/tools/licenses.py
@@ -550,7 +550,7 @@ def FindThirdPartyDirs(prune_paths, root, extra_third_party_dirs=None):
       third_party_dirs.add(dir)
       ProcessAdditionalReadmePathsJson(root, dir, third_party_dirs)
 
-  return third_party_dirs
+  return sorted(third_party_dirs)
 
 
 def FindThirdPartyDirsWithFiles(root):
@@ -601,7 +601,7 @@ def GetThirdPartyDepsFromGNDepsOutput(gn_deps, target_os):
       # Skip over files that are known not to be used on iOS.
       continue
     third_party_deps.add(third_party_path[:-1])
-  return third_party_deps
+  return sorted(third_party_deps)
 
 
 def FindThirdPartyDeps(gn_out_dir, gn_target, target_os):
